Storage/src/Connection/Rest.php

```diff
@@ -431,7 +431,11 @@ private function resolveUploadOptions(array $args)
         }
 
         $args['metadata']['name'] = $args['name'];
+        if (isset($args['retention'])) {
+            $args['metadata']['retention'] = $args['retention'];
```

Comment:
`retention` seems like a request field. See here: https://cloud.google.com/storage/docs/json_api/v1/objects/insert

Why are we passing it in `metadata`?